tof/models.py

```diff
@@ -98,26 +75,69 @@ class Meta:
 
     name = models.CharField(_('Field name'), max_length=250, help_text=_('Name field'))
     title = models.CharField(_('User field name'), max_length=250, help_text=_("Name user's field"))
-    content_type = models.ForeignKey(ContentType, limit_choices_to=~Q(app_label='tof'), on_delete=models.CASCADE)
+    content_type = models.ForeignKey(ContentType, limit_choices_to=~Q(app_label='tof'), on_delete=models.CASCADE, related_name='translatablefields')
 
     def __str__(self):
         return f'{self.content_type.model}|{self.title}'
 
     def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
-        prepare_cls_for_translate(self.content_type.model_class(), self)
+        self.add_translation_to_class()
 
     def delete(self, *args, **kwargs):
-        cls = self.content_type.model_class()
-        ct_pk = self.content_type.pk
-        name = self.name
         super().delete(*args, **kwargs)
+        self.remove_translation_from_class()
```

Comment:
It do not work after delete because self.id is None

Reply:
fix